### DISCUSSION TITLE
SAML Integration in DefectDojo 2.27.0

### DISCUSSION THREAD

**Discussion by Elgrandeothman** [DefectDojo/django-DefectDojo#9040] [Q&A]
We are in the process of integrating the SAML authentication method into our DefectDojo instance. To achieve this, we need to establish a new SAML partnership with the Identity Provider, where we will provide specific information from the DefectDojo side. Does anyone have suggestions for what should be included in the '**Your application metadata (URL or .XML file):**' section?

**Comment by rsantacreu-ust** [comment]
Hi,

I'm also trying to achive saml integration, in my case with AWS Identity Center, but I have some questions, did you achieve integration with SAML?

I have set in my values.yaml from my helm chart the following variables

extraConfigs: 
    DD_SAML2_ENABLED: 'true'
    DD_SAML2_LOGIN_BUTTON_TEXT: 'Login with SAML'
    DD_SAML2_METADATA_AUTO_CONF_URL: 'url_identity_center'
    DD_SAML2_ATTRIBUTES_MAP: "'Email' = 'email', 'Username' = 'username', 'First_name' =  'first_name',  'Lastname' = 'last_name'"

but I'm getting error 403 in AWS. In defect dojo SAML documentation not very clear about the urls of SAML that I should use on AWS idp

[defectdojo-saml](https://defectdojo.github.io/django-DefectDojo/integrations/social-authentication/#saml-20)

I'm using this defectdojo url for Application ACS URL and Application SAML audience on my idp

https://defectdojo.tools.mydomain.com/saml2/login

I'll be really helpful if anyone could help me with this

Thanks
Best regards!

**Reply by Elgrandeothman** [reply]
Hello,
If you are still encountering error 403, it may be due to one of two reasons:
-The first possibility is that you are attempting to connect with SAML, but there is no existing user, resulting in denied access. To test this scenario, set the variable DD_SAML2_CREATE_USER to True.
-The second reason could be that your attribute mapping is incorrect. Verify the attributes that your IdP sends in the SAML response. Additionally, consider using the DD_SAML2_ALLOW_UNKNOWN_ATTRIBUTE variable, as it might be helpful in some cases.

You can check the dojo/settings/settings.dist.py file where you can find  additional options for customizing your SAML configuration.
